docs/src/adr/011adr-smt-arrays.md

```diff
@@ -169,11 +169,112 @@ The following changes will be made to implement the new encoding of sets:
     (assert (= (select set_1 7) false)) ; Without the list we would query set_3 here 
     ```
 
-## 4. Encoding tuples and records
+## 4. Encoding functions and sequences
 
-TODO
+Functions are currently encoded as sets of pairs, with each pair representing a mapping present in
+the function. The first element of a pair is a tuple containing some function arguments and the second
+element is the return value given by such arguments. The handling of functions is thus given by 
+operations over sets and tuples. Sequences of type `T` are currently encoded as tuples of form
+`⟨start,end,fun⟩`, where `start` and `end` are integers and `fun` is a function from integers to `T`.
+The new encoding of functions will thus encompass sequences, as their tuple representations is
+intended to be kept.
+
+The new encoding will, like the current one, also map tuples of arguments to return values, but
+will do so natively  instead of simply relying on sets. A function will be represented by two SMT
+arrays. The first array will store the domain of the function and will be encoded as a standard 
+TLA+ set. The second array will store the mappings, having sort `<S1,...,Sn>` as its domain, with
+`Si` being the sort of argument `i`, and the sort of the function's codomain as its range. 
+The sorts of the array domain and range can be infinite, but the domain of the function itself,
+and by implication the number of mappings, will always be finite.
```

Comment:
I guess, the number of mappings may be infinite, if a function maps to integers

Reply:
I don't know how this can be possible. Doesn't the number of mappings equal the cardinality of the function's domain? If so, and if we are not mapping from an infinite set, the number of mappings should be finite.